File: component-service/src/main/java/dev/yidafu/pan/component/domain/dao/ComponentDynamicSqlSupport.java
package dev.yidafu.pan.component.domain.dao;

import java.sql.JDBCType;
import java.util.Date;
import javax.annotation.Generated;
import org.mybatis.dynamic.sql.AliasableSqlTable;
import org.mybatis.dynamic.sql.SqlColumn;

public final class ComponentDynamicSqlSupport {
    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final Component component = new Component();

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<Long> id = component.id;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<Long> screenId = component.screenId;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<Long> groupId = component.groupId;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<String> name = component.name;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<String> layerName = component.layerName;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<Byte> isGroup = component.isGroup;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<Integer> width = component.width;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<Integer> height = component.height;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<Integer> offsetX = component.offsetX;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<Integer> offsetY = component.offsetY;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<Integer> zIndex = component.zIndex;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<Byte> isLock = component.isLock;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<Byte> isLockAspectRatio = component.isLockAspectRatio;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<String> category = component.category;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<String> subCategory = component.subCategory;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<String> umdJsUrl = component.umdJsUrl;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<Date> createdAt = component.createdAt;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<Date> updatedAt = component.updatedAt;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<String> styleLabelConfig = component.styleLabelConfig;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<String> styleConfig = component.styleConfig;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<String> requestConfig = component.requestConfig;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final SqlColumn<String> interactConfig = component.interactConfig;

    @Generated("org.mybatis.generator.api.MyBatisGenerator")
    public static final class Component extends AliasableSqlTable<Component> {
        public final SqlColumn<Long> id = column("id", JDBCType.BIGINT);

        public final SqlColumn<Long> screenId = column("screen_id", JDBCType.BIGINT);

        public final SqlColumn<Long> groupId = column("group_id", JDBCType.BIGINT);

        public final SqlColumn<String> name = column("name", JDBCType.VARCHAR);

        public final SqlColumn<String> layerName = column("layer_name", JDBCType.VARCHAR);

        public final SqlColumn<Byte> isGroup = column("is_group", JDBCType.TINYINT);

        public final SqlColumn<Integer> width = column("width", JDBCType.INTEGER);

        public final SqlColumn<Integer> height = column("height", JDBCType.INTEGER);

        public final SqlColumn<Integer> offsetX = column("offset_x", JDBCType.INTEGER);

        public final SqlColumn<Integer> offsetY = column("offset_y", JDBCType.INTEGER);

        public final SqlColumn<Integer> zIndex = column("z_index", JDBCType.INTEGER);

        public final SqlColumn<Byte> isLock = column("is_lock", JDBCType.TINYINT);

        public final SqlColumn<Byte> isLockAspectRatio = column("is_lock_aspect_ratio", JDBCType.TINYINT);

        public final SqlColumn<String> category = column("category", JDBCType.VARCHAR);

        public final SqlColumn<String> subCategory = column("sub_category", JDBCType.VARCHAR);

        public final SqlColumn<String> umdJsUrl = column("umd_js_url", JDBCType.VARCHAR);

        public final SqlColumn<Date> createdAt = column("created_at", JDBCType.TIMESTAMP);

        public final SqlColumn<Date> updatedAt = column("updated_at", JDBCType.TIMESTAMP);

        public final SqlColumn<String> styleLabelConfig = column("style_label_config", JDBCType.LONGVARCHAR);

        public final SqlColumn<String> styleConfig = column("style_config", JDBCType.LONGVARCHAR);

        public final SqlColumn<String> requestConfig = column("request_config", JDBCType.LONGVARCHAR);

        public final SqlColumn<String> interactConfig = column("interact_config", JDBCType.LONGVARCHAR);

        public Component() {
            super("vs_components", Component::new);
        }
    }
}

File: component-service/src/main/kotlin/dev/yidafu/pan/component/domain/mapper/ComponentDynamicSqlSupport.kt
package dev.yidafu.pan.component.domain.mapper

import java.sql.JDBCType
import java.util.Date
import org.mybatis.dynamic.sql.AliasableSqlTable
import org.mybatis.dynamic.sql.util.kotlin.elements.column

object ComponentDynamicSqlSupport {
    val component = Component()

    val id = component.id

    val screenId = component.screenId

    val groupId = component.groupId

    val name = component.name

    val layerName = component.layerName

    val isGroup = component.isGroup

    val width = component.width

    val height = component.height

    val offsetX = component.offsetX

    val offsetY = component.offsetY

    val zIndex = component.zIndex

    val isLock = component.isLock

    val isLockAspectRatio = component.isLockAspectRatio

    val category = component.category

    val subCategory = component.subCategory

    val umdJsUrl = component.umdJsUrl

    val createdAt = component.createdAt

    val updatedAt = component.updatedAt

    class Component : AliasableSqlTable<Component>("vs_component", ComponentDynamicSqlSupport::Component) {
        val id = column<Long>(name = "id", jdbcType = JDBCType.BIGINT)

        val screenId = column<Long>(name = "screen_id", jdbcType = JDBCType.BIGINT)

        val groupId = column<Long>(name = "group_id", jdbcType = JDBCType.BIGINT)

        val name = column<String>(name = "name", jdbcType = JDBCType.VARCHAR)

        val layerName = column<String>(name = "layer_name", jdbcType = JDBCType.VARCHAR)

        val isGroup = column<Byte>(name = "is_group", jdbcType = JDBCType.TINYINT)

        val width = column<Int>(name = "width", jdbcType = JDBCType.INTEGER)

        val height = column<Int>(name = "height", jdbcType = JDBCType.INTEGER)

        val offsetX = column<Int>(name = "offset_x", jdbcType = JDBCType.INTEGER)

        val offsetY = column<Int>(name = "offset_y", jdbcType = JDBCType.INTEGER)

        val zIndex = column<Int>(name = "z_index", jdbcType = JDBCType.INTEGER)

        val isLock = column<Byte>(name = "is_lock", jdbcType = JDBCType.TINYINT)

        val isLockAspectRatio = column<Byte>(name = "is_lock_aspect_ratio", jdbcType = JDBCType.TINYINT)

        val category = column<String>(name = "category", jdbcType = JDBCType.VARCHAR)

        val subCategory = column<String>(name = "sub_category", jdbcType = JDBCType.VARCHAR)

        val umdJsUrl = column<String>(name = "umd_js_url", jdbcType = JDBCType.VARCHAR)

        val createdAt = column<Date>(name = "created_at", jdbcType = JDBCType.TIMESTAMP)

        val updatedAt = column<Date>(name = "updated_at", jdbcType = JDBCType.TIMESTAMP)
    }
}